libs/blocks/merch-card/merch-card.js

```diff
@@ -367,6 +367,68 @@ const addStock = (merchCard, styles) => {
   }
 };
 
+
+const addAddon = (merchCard, styles) => {
+  if (styles.includes('add-addon') && (merchCard.variant === MINI_COMPARE_CHART || merchCard.variant === PRODUCT)) {
+    let selector;
+    if (styles.includes('edu')) {
+        selector = '.merch-offers.edu';
+    } else if (styles.includes('team')) {
+        selector = '.merch-offers.team';
+    } else {
+        selector = '.merch-offers:not(.edu):not(.team)';
+    }
+    let planType = '';
+    if (styles.includes('m2m')) {
+    planType = 'M2M';
+    } else if (styles.includes('abm')) {
+    planType = 'ABM';
+    } else if (styles.includes('puf')) {
+    planType = 'PUF';
+    }
+
+if (planType) {
+    merchCard.setAttribute('plan-type', planType);
+}
+    const merchOffers = document.querySelector(selector);
+    const addonLabel = merchOffers?.querySelector('p');
+    const addonLabelCopy = addonLabel?.cloneNode(true);
+    const strongTag = createTag('strong', '', addonLabelCopy.innerHTML);
+    addonLabelCopy.replaceChildren(strongTag);
+    const addonDescriptions = [...merchOffers?.querySelectorAll('p:not(:first-child)')]
+        .filter(p => p.innerText.trim())
+        .map((p, i) => createTag(
+            'span',
+            '',
+            `${i === 0 ? '' : ' '}${p.cloneNode(true).innerHTML}${i === 0 ? ' ' : ''}`
+        ));
+    const addon = createTag('merch-addon', { slot: 'addon'});
+    merchCard.appendChild(addon);
+    addon.appendChild(addonLabelCopy);
+    const spans = merchOffers?.querySelectorAll('span[data-wcs-osi]');
+    spans?.forEach(span => {
+      const p = createTag('p', { 'data-plan-type': '' });
```

Comment:
you don't need to handle plan types on `merch-addon`, the latter handles it automatically